cmd/skaffold/app/cmd/render.go

```diff
@@ -48,13 +48,14 @@ func NewCmdRender() *cobra.Command {
 			{Value: &renderFromBuildOutputFile, Name: "build-artifacts", Shorthand: "a", Usage: "File containing build result from a previous 'skaffold build --file-output'"},
 			{Value: &offline, Name: "offline", DefValue: false, Usage: `Do not connect to Kubernetes API server for manifest creation and validation. This is helpful when no Kubernetes cluster is available (e.g. GitOps model). No metadata.namespace attribute is injected in this case - the manifest content does not get changed.`, IsEnum: true},
 			{Value: &renderOutputPath, Name: "output", Shorthand: "o", DefValue: "", Usage: "file to write rendered manifests to"},
-			{Value: &opts.DigestSource, Name: "digest-source", DefValue: "local", Usage: "Set to 'local' to build images locally and use digests from built images; Set to 'remote' to resolve the digest of images by tag from the remote registry; Set to 'none' to use tags directly from the Kubernetes manifests. Set to 'tag' to use tags directly from the build.", IsEnum: true},
+			{Value: &opts.DigestSource, Name: "digest-source", DefValue: "remote", Usage: "Set to 'remote' to skip builds and resolve the digest of images by tag from the remote registry. Set to 'local' to build images locally and use digests from built images. Set to 'tag' to use tags directly from the build. Set to 'none' to use tags directly from the Kubernetes manifests.", IsEnum: true},
 		}).
 		WithHouseKeepingMessages().
 		NoArgs(doRender)
 }
 
 func doRender(ctx context.Context, out io.Writer) error {
+	opts.RenderOnly = true
```

Comment:
Are we going to do away with `RenderOnly`?  Is it worth using the `DefValuePerCommand` in `flags.go`?

It seems a good idea that values in `opts` should have flags in `flags.go`

Reply:
we're doing away with it as a flag, but we still need to know at **build time** if we're rendering, to know what to do with the digest.

in general I agree that we should have corresponding flags for values in `opts`, but in this case since we have two overlapping code paths we need to have a switch that gets passed down from the `cmd` level. the run context seems like a convenient way to do this....though maybe it's bad code style, and we should just use a param to `Build` instead.

in this case I think we should leave this as is, since this flag is being removed soon anyway. when it gets removed I'll change the name and see if passing a `bool` to `Build` is a cleaner way to switch on the branching code path when I do that.